### ISSUE TITLE
Video streams info does not show drm-format yet (new in Gst 1.24)

### ISSUE THREAD

**Issue by rmader** [Rafostar/clapper#428]
Congratulations to and thanks for the awesome new release!

Just gave it a try and saw that the video stream info panel is missing a small detail: when the new special format `DMA_DRM` is used - which is/will basically always be the case with hardware decoding, then the actual format can be found in the `drm-format` field. And that one is not yet displayed in the stream info. Maybe it can be added at some point :)

![image](https://github.com/Rafostar/clapper/assets/6313774/40e86524-9d2a-4c63-9cc8-51692ab40555)

**Comment by Rafostar** [comment]
Yeah. The whole DMABufs + DRM modifiers formats related stuff was skipped in this release. It was mostly written against GStreamer 1.22 (cause it took me this long to rewrite stuff with a shared GTK widget functionality).

Supporting DMABufs, gpu offload with DRM negotiation will by my target for next stable (even number) release.

**Comment by rmader** [comment]
I see, thanks a lot. Feel free to close the issue / make it a duplicate or so.

**Comment by Rafostar** [comment]
They are related, but I feel this is indeed worth a different issue, so I will not forget to update "info" window. :smile:

**Comment by rmader** [comment]
@Rafostar Hi! I shortly wanted to ping you on this issue and drop a bit of info why I think it would be great to move this forward:
1. GTK and Gstreamer DMABuf handling has matured quite a bit over the last years. Crucially as of the new Gst 1.28 not only HW decoded video allows zero-copy import, but software decoders as well. On most SoCs / devices with unified memory that allows direct sharing of buffer between the the decoder and the GPU, making us among best in class regarding performance, especially in memory bandwidth limited scenarios (which is often the case).
2. That (u)dmabuf work already helps Clapper as it's now (with Gst 1.28) usually uses `Dmabuf` or `DirectDmabufExternal` in `glupload`, removing one copy in the `GLMemory` uploader - but especially when using the GTK Vulkan renderer that's still quite a bit slower than directly importing the buffer into the renderer (and the GL<->VK dance is generally a bit unfortunate).
3. Directly importing into GTK also allows to properly display HDR video nowadays - something `glsinkbin` unfortunately still doesn't do and possible never will.

Given that Clapper is now quite widely used in the Gnome ecosystem (great to see, congrats!), I'd love to see that optimizations coming to it as well.

IIUC, this should mostly come down to have a new handler similar to `gst_clapper_gl_context_handler_make_gl_texture()`, but building the texture with `GdkDmabufTextureBuilder` like [here](https://gitlab.gnome.org/GNOME/gtk/-/blob/81cec9fcbe576455effd3e6a1c3b5d1bb700b097/gtk/media/gtkgstsink.c#L651-711). On top of that it should mainly need some caps work, ensuring to query the supported formats from GTK and passing them on like [here](https://gitlab.gnome.org/GNOME/gtk/-/blob/81cec9fcbe576455effd3e6a1c3b5d1bb700b097/gtk/media/gtkgstsink.c#L248).

Is that something you might be interested and have time to look into this cycle? 🙏  If not, I'll try to give it a shot myself - but would you then be interested in reviewing and guiding me a bit?

Regards

**Comment by Rafostar** [comment]
Hi, do not worry. I will get into it... eventually 😄 

> If not, I'll try to give it a shot myself

Please do not waste your time on that. Current Clapper sink needs major reworking anyway to support that. This was postponed due to to these new DMABuf related APIs in both GStreamer and GTK changing constantly (and multiple MESA issues on early stages).

Thanks for the offer to help through 👍
